### PR TITLE
CI: update github/super-linter docker tag to v4

### PR DESCRIPTION
**Pull request by neteler**
Replaces #2778

Selectively update `super-linter.yml` to GRASS GIS Addons PR https://github.com/OSGeo/grass-addons/pull/847

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [github/super-linter](https://togithub.com/github/super-linter) | docker | major | `v2.2.2` -> `v4` |

### CODE REVIEW

**Comment by neteler** [comment]
In  baea7da my attempt to fix

`[FATAL]   Behavior not supported, please either only include (VALIDATE=true) or exclude (VALIDATE=false) linters, but not both`

File: .github/workflows/super-linter.yml [hunk 1]
```diff
@@ -19,20 +19,21 @@ jobs:
...
-          VALIDATE_CSS: false
-          VALIDATE_DOCKER: false
+          # VALIDATE_BASH: true
+          # VALIDATE_CSS: true
+          # VALIDATE_DOCKER: true
           VALIDATE_JAVASCRIPT_ES: true
-          VALIDATE_JAVASCRIPT_STANDARD: false
+          # VALIDATE_JAVASCRIPT_STANDARD: true
           VALIDATE_JSON: true
           VALIDATE_MD: true
```
Comment (nilason) [inline .github/workflows/super-linter.yml:36]:
```suggestion
          VALIDATE_MARKDOWN: true
```

According to https://github.com/github/super-linter#environment-variables. After #2762 they should all pass too.

**Review by nilason — APPROVED** [review]

**Comment by echoix** [comment]
LGTM! (Looks good to me!)

File: .github/workflows/super-linter.yml [hunk 2]
```diff
@@ -19,20 +19,21 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - name: Lint code base
-        uses: docker://github/super-linter:v2.2.2
+        uses: github/super-linter@v4
         env:
-          # Listed but disabled linters would be nice to have.
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (nilason) [inline .github/workflows/super-linter.yml:24]: FYI: setting `GITHUB_TOKEN` was needed for listing each test:

Screenshot

Comment (neteler) [reply .github/workflows/super-linter.yml:24]: Oh, I see. I assume you have updated this?

Comment (nilason) [reply .github/workflows/super-linter.yml:24]: Only the markdown files with #2762. The others were apparently good.

Comment (nilason) [reply .github/workflows/super-linter.yml:24]: Before this upgrade of super-linter (incl. GITHUB_TOKEN), it **only** looked like:

Screenshot2

which wasn't very informative.

Comment (neteler) [reply .github/workflows/super-linter.yml:24]: > FYI: setting `GITHUB_TOKEN` was needed for listing each test:

Ah, you meant adding the `GITHUB_TOKEN`  line in the super-linteryml file, right? (I initially thought that you have set the secret somewhere in the GH settings.)

Comment (nilason) [reply .github/workflows/super-linter.yml:24]: See beginning of https://github.com/github/super-linter#how-to-use

> NOTE: If you pass the Environment variable GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} in your workflow, then the GitHub Super-Linter will mark the status of each individual linter run in the Checks section of a pull request. Without this you will only see the overall status of the full run. There is no need to set the GitHub Secret as it is automatically set by GitHub, it only needs to be passed to the action.

**Pull request merged**